geode-core/src/main/java/org/apache/geode/management/internal/beans/ManagementAdapter.java

```diff
@@ -719,6 +812,7 @@ public void handleCacheServerStop(CacheServer server) {
    * @param cache GemFire Cache instance. For now client cache is not supported
    */
   public void handleCacheRemoval(Cache cache) throws ManagementException {
+    readWriteLock.writeLock().lock();
```

Comment:
Super nit-picky, but I'd prefer it if this were referenced with `this.` to be consistent with (most of) the other field-references in this method.  Or better yet, for the others `this.`s to be removed to be consistent with (most of) the rest of the class.